src/ripple/basics/Slice.h

```diff
@@ -120,6 +125,27 @@ Stream& operator<<(Stream& s, Slice const& v)
     return s;
 }
 
+template <class T, class Alloc>
+std::enable_if_t<
+    std::is_same<T, char>::value ||
+        std::is_same<T, unsigned char>::value,
```

Comment:
It's a shame there's no way to tell if an iterator points to contiguous memory or we could write a generic `make_Slice` for anything that `std::begin` is contiguous and references an element of sizeof(1).